xapian-core/weight/tfidfweight.cc

```diff
@@ -184,21 +184,22 @@ TfIdfWeight::get_wdfn(Xapian::termcount wdf, Xapian::termcount doclen,
 {
     switch (c) {
 	case 'b':
-	    if (wdf == 0) return 0;
+	    if (wdf == 0) return 1.0;
 	    return 1.0;
 	case 's':
+	    if (wdf == 0) return 1.0;
 	    return (wdf * wdf);
 	case 'l':
-	    if (wdf == 0) return 0;
+	    if (wdf == 0) return 1.0;
 	    return (1 + log(double(wdf)));
```

Comment:
Without the current check not sure of `wdf==0`. Implementation before looks wrong as log(0) would be -inf.

```
-inf
```